src/Std/Time.lean

```diff
@@ -0,0 +1,219 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Time.Time
+import Std.Time.Date
+import Std.Time.Zoned
+import Std.Time.Format
+import Std.Time.DateTime
+import Std.Time.Notation
+import Std.Time.Duration
+import Std.Time.Zoned.Database
+
+namespace Std
+namespace Time
+
+/-!
+# Time
+
+The Lean API for date, time, and duration functionalities.
```

Comment:
In this overview there should be a detailed explanation about DST-aware datetime arithmetic, how `ZonedDateTime` carries around a `ZoneRules`, how to obtain a `ZoneRules`, and how to opt-out of DST-aware datetime arithmetic by working with `TimeZone` instead.

Reply:
I added the explanation in the `ZonedDateTime` part.